src/gtest/test_arith_uint256.cpp

```diff
@@ -0,0 +1,567 @@
+    EXPECT_TRUE((R2L * 0x87654321UL).ToString() == "23f7816e30c4ae2017257b7a0fa64d60402f5234d46e746b61c960d09a26d070");
+}
+
+TEST(arith_uint256_tests, divide)
+{
+    arith_uint256 D1L("AD7133AC1977FA2B7");
+    arith_uint256 D2L("ECD751716");
+    EXPECT_TRUE((R1L / D1L).ToString() == "00000000000000000b8ac01106981635d9ed112290f8895545a7654dde28fb3a");
+    EXPECT_TRUE((R1L / D2L).ToString() == "000000000873ce8efec5b67150bad3aa8c5fcb70e947586153bf2cec7c37c57a");
+    EXPECT_TRUE(R1L / OneL == R1L);
+    EXPECT_TRUE(R1L / MaxL == ZeroL);
+    EXPECT_TRUE(MaxL / R1L == 2);
+    EXPECT_THROW(R1L / ZeroL, uint_error);
+    EXPECT_TRUE((R2L / D1L).ToString() == "000000000000000013e1665895a1cc981de6d93670105a6b3ec3b73141b3a3c5");
+    EXPECT_TRUE((R2L / D2L).ToString() == "000000000e8f0abe753bb0afe2e9437ee85d280be60882cf0bd1aaf7fa3cc2c4");
+    EXPECT_TRUE(R2L / OneL == R2L);
+    EXPECT_TRUE(R2L / MaxL == ZeroL);
+    EXPECT_TRUE(MaxL / R2L == 1);
+    EXPECT_THROW(R2L / ZeroL, uint_error);
+}
+
+
+bool almostEqual(double d1, double d2)
+{
+    return fabs(d1-d2) <= 4*fabs(d1)*std::numeric_limits<double>::epsilon();
+}
+
+TEST(arith_uint256_tests, methods) // GetHex SetHex size() GetLow64 GetSerializeSize, Serialize, Unserialize
+{
+    EXPECT_TRUE(R1L.GetHex() == R1L.ToString());
+    EXPECT_TRUE(R2L.GetHex() == R2L.ToString());
+    EXPECT_TRUE(OneL.GetHex() == OneL.ToString());
+    EXPECT_TRUE(MaxL.GetHex() == MaxL.ToString());
+    arith_uint256 TmpL(R1L);
+    EXPECT_TRUE(TmpL == R1L);
+    TmpL.SetHex(R2L.ToString());   EXPECT_TRUE(TmpL == R2L);
+    TmpL.SetHex(ZeroL.ToString()); EXPECT_TRUE(TmpL == 0);
+    TmpL.SetHex(HalfL.ToString()); EXPECT_TRUE(TmpL == HalfL);
+
+    TmpL.SetHex(R1L.ToString());
+    EXPECT_TRUE(R1L.size() == 32);
+    EXPECT_TRUE(R2L.size() == 32);
+    EXPECT_TRUE(ZeroL.size() == 32);
+    EXPECT_TRUE(MaxL.size() == 32);
+    EXPECT_TRUE(R1L.GetLow64()  == R1LLow64);
+    EXPECT_TRUE(HalfL.GetLow64() ==0x0000000000000000ULL);
+    EXPECT_TRUE(OneL.GetLow64() ==0x0000000000000001ULL);
+
+    for (unsigned int i = 0; i < 255; ++i)
+    {
+        EXPECT_TRUE((OneL << i).getdouble() == ldexp(1.0,i));
+    }
+    EXPECT_TRUE(ZeroL.getdouble() == 0.0);
+    for (int i = 256; i > 53; --i)
+        EXPECT_TRUE(almostEqual((R1L>>(256-i)).getdouble(), ldexp(R1Ldouble,i)));
+    uint64_t R1L64part = (R1L>>192).GetLow64();
+    for (int i = 53; i > 0; --i) // doubles can store all integers in {0,...,2^54-1} exactly
+    {
+        EXPECT_TRUE((R1L>>(256-i)).getdouble() == (double)(R1L64part >> (64-i)));
+    }
+}
+
+TEST(arith_uint256_tests, bignum_SetCompact)
+{
+    arith_uint256 num;
+    bool fNegative;
+    bool fOverflow;
+    num.SetCompact(0, &fNegative, &fOverflow);
+    EXPECT_EQ(num.GetHex(), "0000000000000000000000000000000000000000000000000000000000000000");
+    EXPECT_EQ(num.GetCompact(), 0U);
+    EXPECT_EQ(fNegative, false);
```

Comment:
EXPECT_EQ => EXPECT_FALSE,
all fNegative, fOverflow checks below... some of them EXPECT_TRUE, some EXPECT_FALSE